### ISSUE TITLE
Adding VM to Domain

### ISSUE THREAD

**Issue by kalalvishal** [pleasereleaseme/Infrastructure#1]
Hi,

Thanks for the JSON. I am following your JSON in my test environment to deploy VM and add to the domain, VM is spinning up fine but while adding to Domain, its failed saying extension not found. Could you please help me and guide how to add Domain Join extension. Thank you very much.

**Comment by GrahamDSmith** [comment]
Have you seen the accompanying blog post:

[Continuous Delivery with TFS / VSTS – Join a VM to a Domain with Azure Resource Manager Templates](http://pleasereleaseme.net/continuous-delivery-tfs-vsts-join-vm-domain-azure-resource-manager-templates/)

Hope it helps!

**Comment by kalalvishal** [comment]
Yes, i have seen the post. I have created the extension using power-shell and able to join the system using adding extensions from Powershell but not able to add using JSON.

**Comment by GrahamDSmith** [comment]
It's been a while since I worked on this so I'm not in a good position to help I'm afraid. I would start by using hard-coded values rather than variables and double-check all the details for connecting to your domain. Also have a look at:

https://github.com/Azure/azure-quickstart-templates/tree/master/201-vm-domain-join

as I see that (for example) the extension version has changed so my code is now out of date.
